specs/_features/maxeb_increase/capella.py

```diff
@@ -3721,54 +3685,39 @@ def get_expected_withdrawals(state: BeaconState) -> Sequence[Withdrawal]:
     validator_index = state.next_withdrawal_validator_index
     withdrawals: List[Withdrawal] = []
     consumed = 0
-    for withdrawal in state.eligible_slashing_withdrawals:
-        if withdrawal.withdrawable_epoch > epoch or consumed == MAX_WITHDRAWALS_PER_PAYLOAD:
+    for withdrawal in state.pending_partial_withdrawals:
+        if withdrawal.withdrawable_epoch > epoch or len(withdrawals) == MAX_WITHDRAWALS_PER_PAYLOAD // 2:
```

Comment:
why limit to half of max withdrawals ? dont want only partials to fill up the list?

Reply:
We can have a separate constant for this limitation. As long as exits remain in the sweep which I see like a good compromise between the engineering complexity and design goals of partial withdrawals, we want to limit a number of partial withdrawals to let full exits be processed.